Box.V2/Extensions/BoxExtensions.cs

```diff
@@ -34,5 +34,13 @@ internal static string ThrowIfNullOrWhiteSpace(this string value, string name)
 
             return value;
         }
+
+        internal static T? ThrowIfNullEnum<T>(this T? param, string name)  where T : struct
```

Comment:
Why could not we remove the "where T : class" on ThrowIfNull(), so that we don't need ThrowIfNullEnum?